src/open_clip/hf_model.py

```diff
@@ -48,10 +49,22 @@ class PreTrainedTextEncoder(nn.Module):
     def __init__(
             self, 
             model_name_or_path:str,
-            embedding_dim:int,
+
+            context_length:int = 77,
+            vocab_size:int = 49408,
+            width:int = 512,
+            heads:int = 8,
+            layers:int = 12,
+            output_dim:int = 512,
```

Comment:
vocab_size, heads, layers is given for pretrained model, so we really only need  an argument for the expected output embedding dim. We can also add an optional `config` input default to none, to enable instantiating untrained HF transformer models. For this scenario different models will require different args, so it's better to pass `config`.

Reply:
right so we want to add a config of type PretrainedConfig and if that argument is passed and hence isn't None we want to instantiate the model like:
```
self.transformer = AutoModel.from_pretrained(config=self.config)
```
yes?

Reply:
yes, this should work

Reply:
actually `from_pretrained` will throw error, there is `_from_config` method that should fit in this case

Reply:
Yup, that's what I used. You can check out the latest changes and LMK if it looks good to merge for you